src/test/java/org/assertj/core/internal/paths/Paths_assertIsDirectoryContaining_with_Predicate_Test.java

```diff
@@ -0,0 +1,137 @@
+import static org.assertj.core.error.ShouldContain.directoryShouldContain;
+import static org.assertj.core.error.ShouldExist.shouldExist;
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+import static org.assertj.core.util.Lists.emptyList;
+import static org.junit.jupiter.api.condition.OS.WINDOWS;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.function.Predicate;
+
+import org.assertj.core.internal.PathsBaseTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+
+/**
+ * @author Valeriy Vyrva
+ */
+class Paths_assertIsDirectoryContaining_with_Predicate_Test extends PathsBaseTest {
+
+  @Test
+  void should_fail_if_filter_is_null() throws IOException {
+    // GIVEN
+    Path actual = Files.createDirectory(tempDir.resolve("actual"));
+    Predicate<Path> filter = null;
+    // WHEN
+    Throwable thrown = catchThrowable(() -> paths.assertIsDirectoryContaining(info, actual, filter));
+    // THEN
+    then(thrown).isInstanceOf(NullPointerException.class)
+                .hasMessage("The paths filter should not be null");
+  }
+
+  @Test
+  void should_fail_if_actual_is_null() {
+    // GIVEN
+    Path actual = null;
+    Predicate<Path> filter = path -> true;
+    // WHEN
+    AssertionError error = expectAssertionError(() -> paths.assertIsDirectoryContaining(info, actual, filter));
+    // THEN
+    then(error).hasMessage(actualIsNull());
+  }
+
+  @Test
+  void should_fail_if_actual_does_not_exist() {
+    // GIVEN
+    Path actual = tempDir.resolve("non-existent");
+    Predicate<Path> filter = path -> true;
+    // WHEN
+    AssertionError error = expectAssertionError(() -> paths.assertIsDirectoryContaining(info, actual, filter));
+    // THEN
+    then(error).hasMessage(shouldExist(actual).create());
+  }
+
+  @Test
+  void should_fail_if_actual_is_not_a_directory() throws IOException {
+    // GIVEN
+    Path actual = Files.createFile(tempDir.resolve("file"));
+    Predicate<Path> filter = path -> true;
+    // WHEN
+    AssertionError error = expectAssertionError(() -> paths.assertIsDirectoryContaining(info, actual, filter));
+    // THEN
+    then(error).hasMessage(shouldBeDirectory(actual).create());
+  }
+
+  @Test
+  @DisabledOnOs(value = WINDOWS, disabledReason = "gh-FIXME")
+  void should_rethrow_IOException_as_UncheckedIOException() throws Exception {
+    // GIVEN
+    Path actual = Files.createDirectory(tempDir.resolve("actual"));
+    Predicate<Path> filter = path -> true;
+    IOException cause = new IOException("boom!");
+    given(nioFilesWrapper.newDirectoryStream(actual, filter)).willThrow(cause);
```

Comment:
`Paths_assertIsDirectoryContaining_with_String_Test#should_rethrow_IOException_as_UncheckedIOException` uses     `willThrow(cause).given(nioFilesWrapper).newDirectoryStream(any(), any());`  and is not disabled on windows, I have not idea why though.

Reply:
It seems we get failures on Windows only when Files.newInputStream is used.

There are also other tests like `Paths_assertIsCanonical_Test#should_rethrow_IOException_as_UncheckedIOException` or `Paths_assertHasSize_Test#should_rethrow_IOException_as_UncheckedIOException` which are working fine on Windows.

Reply:
I overlooked your comment, and there were tests failing and disabled using `newDirectoryStream`. It seems that inverting the stubbing procedure fixed the issue: 57680333d70827eba86c443a10ec3f840b088414

I'll try the same with the other strange failures.